src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -695,6 +711,52 @@ private void determineUnitsToAttackWith(final List<ProTerritory> prioritizedTerr
       Map<Unit, Set<Territory>> sortedUnitAttackOptions =
           tryToAttackTerritories(prioritizedTerritories, alreadyMovedUnits);
 
+      // Clear bombers
+      for (final Territory t : attackMap.keySet()) {
+        attackMap.get(t).getBombers().clear();
+      }
+
+      // Get all units that have already moved
+      final Set<Unit> alreadyAttackedWithUnits = new HashSet<>();
+      for (final Territory t : attackMap.keySet()) {
+        alreadyAttackedWithUnits.addAll(attackMap.get(t).getUnits());
+        alreadyAttackedWithUnits.addAll(attackMap.get(t).getAmphibAttackMap().keySet());
+      }
+
+      // Check to see if any territories can be bombed
+      final Map<Unit, Set<Territory>> bomberMoveMap = territoryManager.getAttackOptions().getBomberMoveMap();
+      for (final Unit unit : bomberMoveMap.keySet()) {
+        if (alreadyAttackedWithUnits.contains(unit)) {
+          continue;
+        }
+        Optional<Territory> maxBombingTerritory = Optional.empty();
+        int maxBombingScore = 3;
+        for (final Territory t : bomberMoveMap.get(unit)) {
+          final boolean territoryCanBeBombed = t.getUnits().someMatch(Matches.UnitCanProduceUnitsAndCanBeDamaged);
+          if (territoryCanBeBombed && canAirSafelyLandAfterAttack(unit, t)) {
+            final int noAABombingDefense = t.getUnits().someMatch(Matches.UnitIsAAforBombingThisUnitOnly) ? 0 : 1;
+            int maxDamage = 0;
+            final TerritoryAttachment ta = TerritoryAttachment.get(t);
+            if (ta != null) {
+              maxDamage = ta.getProduction();
+            }
+            final int numExistingBombers = attackMap.get(t).getBombers().size();
+            final int remainingDamagePotential = maxDamage - 3 * numExistingBombers;
+            final int bombingScore = (1 + 9 * noAABombingDefense) * remainingDamagePotential;
+            if (bombingScore > maxBombingScore) {
+              maxBombingScore = bombingScore;
+              maxBombingTerritory = Optional.of(t);
+            }
+          }
+        }
+        if (maxBombingTerritory.isPresent()) {
+          final Territory t = maxBombingTerritory.get();
+          attackMap.get(t).getBombers().add(unit);
+          sortedUnitAttackOptions.remove(unit);
+          ProLogger.debug("Add bomber (" + unit + ") to " + t);
```

Comment:
Why is that beyond this change? It seems like this change is not quite complete, it has not been revised to be clean, it has not been made modular. Adding in future changes are likely going to be really painful. For example, bomber range is extended when bombers can land on carriers. Adding that in, as is, I don't think would be so pleasant. But also, given the complexity, the chance for bugs is greater than if things were simpler and unit tested.